ライブリの全面改修

Sharoku-haga opened the pull request:
#18 の作業になります。

現状ではファイルを作成しただけです。
作業は日曜日に終了予定にしています。
#17 の作業はこちらに合流させます。

2016/10/24追記
いくつか納得できない部分はありますが、ある程度はできています。
現在コメントをつける作業しております

Commit:
[ehn #16] テクスチャー管理クラス追加

変更点
-TextureManagerクラスの追加
-上記クラス追加に伴い、Textureクラスを削除とVertexクラスを変更
-Zバッファ関連の修正

Commit:
[ehn #18] ライブラリの改修

GamaLibのファイルを作成

review-ninja commented:
<a href="https://app.review.ninja/TeamT-SEN/T-SEN/pull/19" target="_blank"><img src="https://app.review.ninja/69837692/pull/19/badge" alt="ReviewNinja"/>

Commit:
Merge branch 'TextureManager' into GameLib

# Conflicts:
#	Twar/Lib_Class/Lib_Class.vcxproj.filters

Commit:
作成にあたって改めてフィルターを整理

Commit:
XFileManagerクラスの追加

Xファイルを一括で管理するクラス

Commit:
DirectInput関連のクラスの関数を一部修正

-キー確認関数
-マウスの関数にホイールを追加
-ウィンドウプロシージャ関数の中を調整

taishi-matsumura commented:
GitHubにはレビュワーのリストとか「承認しました」みたいな機能が無いので、ReviewNinjaとかいうプラグインを入れてみました。
最初は「Some checks haven’t completed yet」ってなってるかと思いますが、その下の「Detail」ってとこクリックするとReviewNinjaのサイトに飛ぶので、そこでコードレビューしてコメント書いたりできます。
で、問題無くなれば手裏剣のアイコンをクリックしてNinjaStarをAddしてあげるとGitHub側も「All checks have passed」ってなるので安心してmergeできるという仕組みみたいです。

プルリク上のやり取りもGitHubよりは見易いかと思います。
[これ](https://app.review.ninja/KingdomCraft-1/KingdomCraft/pull/22/657aa2f509e84dd35c33b2c4937c7c7e6f57341b...41ccd69b24802a37210fa827d133c2bb2c48dbd5)とか、森本の鬼の指摘の一覧が見れますね。
ちょっとBitbucketっぽくなる感じ。

Commit:
[enh #18] VertexManagerの追加と一部ソースの修正

-Vertexクラスを管理するVertexManagerを追加
-それに伴いVertexクラスを修正
-GraphicsDeviceクラスの必要ないライト関連の関数を消去
-InputKeyクラスのメンバ変数を全部privateに変更

Commit:
[enh #18]DebugFontクラス追加と一部ソースの修正

-デバックで使うフォントであるDebugFontクラス追加
-Fontクラスの一部変更
-DebugTimerクラスのソースを修正

Commit:
[enh #18] GameLibクラス追加

-GameLibクラス追加
-GamaLibクラスを削除→スペルミスしてました
-DebugTimerクラスに関数追加

Commit:
ヘッダーにコメントを追加

Commit:
[enh #18] SoundFileManagerの修正

-SoundFileManagerのmapの登録キーをintに変更
-SoundFileManagerに関数追加
-上記変更に伴い、GameLibクラスの一部変更

Commit:
[enh #18] DirectInput系クラスのenumの見直し

-DirectInput系クラスのenumの変更と修正
-GameLibクラスにボタン状態を表すenum追加
-GameLibをSingltonパターンに変更
-マウス座標を取得する関数を取得形式を変更

Commit:
[enh #18] マウス関連の関数の修正

-InputMouseのホイール関連の関数を修正
-GameLibのマウス関連の関数を修正

Commit:
ヘッダーファイルを整理

-前方宣言に変更

Sharoku-haga commented:
お疲れ様です。
ライブラリの改修作業でずが、終わったのでreviewお願いします。
個人的に気になる点にはコメントを書いてありますので、良い案があれば大歓迎です。
チェックしてよければ手裏剣お願いします。

Commit:
GamaLibクラスにコメント追加

Commit:
ウィンドウ起動を確認

起動しなかったので一部ソースを変更

taishi-matsumura commented on Twar/Lib_Class/DebugTimer.cpp:
```diff
@@ -77,6 +77,25 @@ void DebugTimer::ReStart(std::string  timeName)
 	}
 }
 
+// �v�����ʂ̎擾
```
TEXTマクロとかを使ってなさそうなこの辺のファイルはUTF-8化できそうですね。
文字化けウザいのでちょっとずつでも対応お願いします。 !fix

Sharoku-haga replied:
了解しました。随時UTF-8化進めていきます。

Sharoku-haga replied:
DubugTimer.cppはUTF-8化して、コミットしました。

Commit:
DebugTimer.cppをUTF-8に変換

taishi-matsumura commented:
Lib_Class側のInclude pathとLibrary pathが絶対pathになってます !fix

taishi-matsumura commented:
プロジェクトの依存関係を設定して下さい。
現状、Twar側を先にビルドしようとするので、初回はLib_Class.libが無くてエラーになります。
libの方に変更あった場合も、ソリューションを2回ビルドしないとTwar側に最新のlibがリンクされないのでマズいです。 !fix

taishi-matsumura commented on Twar/Lib_Class/DebugTimer.cpp:
```diff
@@ -77,6 +77,25 @@ void DebugTimer::ReStart(std::string  timeName)
 	}
 }
 
+// �v�����ʂ̎擾
```
!fixed

Commit:
プロジェクト依存関係設定完了

taishi-matsumura commented on Twar/Lib_Class/Lib_Class.vcxproj:
```diff
@@ -72,34 +72,42 @@
   </ItemDefinitionGroup>
   <ItemGroup>
     <ClInclude Include="Camera.h" />
+    <ClInclude Include="DebugFont.h" />
```
Lib_Class側のInclude pathとLibrary pathが絶対pathになってます。
あと、その辺りのDebug/Release共通の設定は「すべての構成」に対して設定してやって下さい。 !fix

taishi-matsumura replied:
!fixed

Commit:
Lib_Class側を相対pathに変更

Sharoku-haga commented:
Lib_Class側のPath変更とプロジェクト依存関係の設定しました。
すません。Lib_Class側のパスは完全に見落としていました。

taishi-matsumura commented on Twar/Lib_Class/Lib_Class.vcxproj.filters:
```diff
@@ -1,138 +1,168 @@
-  <ItemGroup>
-    <ClCompile Include="Texture.cpp">
-      <Filter>Lib_Class\DirectXGraphics\Texture</Filter>
+    <ClCompile Include="GraphicsDevice.cpp">
```
フィルタを細かく設定する前に実ファイル/フォルダの構成を作って下さい。
フィルタはファイルシステム上の木構造を表現する程度にしておいた方が吉です。
includeファイルの相対path指定等あちこち変更が必要になると思うので、UTF-8化と併せて少しずつやっつけていきましょか。 !fix

Sharoku-haga replied:
了解しました。
フォルダ構成作成後、ファイルごとにincludeファイルの相対path指定変更とUTF-8化
という工程で随時進めていきます。

Commit:
Fontクラスをフォルダ整理。UTF-8化

Commit:
フィルタ整理

Commit:
Cameraクラスをフォルダ分けとUTF-8化

Commit:
LightクラスをUTF-8化

Commit:
LibclassのいつくつかクラスをUTF-8化

Commit:
Xファイル関連クラスのUTF-8化

Commit:
Lib_ClassとmainのUTF-8化完了

Commit:
実フォルダ整理とソースの一部修正

-GameLibファオルダ内にLibフォルダを作成しファイルを整理
-LightクラスとCameraクラスのコンストラクタにデバイスの引数追加
-GameLibクラスのhのprivateメンバーの位置の変更
-Lib_Class.hの整理

taishi-matsumura commented on Twar/Lib_Class/Camera/Camera.h:
```diff
@@ -0,0 +1,36 @@
+﻿/**
```
doxygenコメント入れてくれてるみたいなので、ついでに。

[ここ](http://cube.atnifty.com/~suzume/wiki/?develop%2Fdoxygen)とかを参考に、doxygenで生成したhtmlファイルをWindowsのヘルプファイル(chm)形式に変換すると捗ります。
読む人向けにchmを、作りたい人向けにdoxyfileを、それぞれリポジトリに入れておくと良いです。(あの大量の.jsとか.htmlとかは不要)
doxyfileは誰のPCでも動くように、各種pathは絶対pathでは無く相対pathで書くように注意。

Sharoku-haga replied:
了解しました。
作業に入ります。

Sharoku-haga commented:
UTF-8化と実フォルダ分け完了しました。
・UTF-8化
reviewninjaでコメントが日本語になっているのを確認いたしました。
別プログラムで試しながら変えましたので、動作も問題ないです。
後、念の為マージテストしてみましたが、文字化けることなくマージもできました。（MergeTestブランチ）
・実フォルダ分け
実フォルダ分けに関してはまず大きくCamera,Light,Font,GameLibで4つに分けております。
更にGameLibフォルダの中でLibフォルダを作成するという形で分けております。

taishi-matsumura commented on Twar/Lib_Class/Lib_Class.vcxproj.filters:
```diff
@@ -1,138 +1,168 @@
-  <ItemGroup>
-    <ClCompile Include="Texture.cpp">
-      <Filter>Lib_Class\DirectXGraphics\Texture</Filter>
+    <ClCompile Include="GraphicsDevice.cpp">
```
気長にボチボチやっていって下さい。 !fixed

taishi-matsumura commented on Twar/Lib_Class/GameLib/GameLib.h:
```diff
@@ -0,0 +1,349 @@
+	* @param[in] Wndproc	ウィンドウプロシージャ関数
+	* @param[in] windowType ウィンドウタイプ trueなら通常,falseならフルスクリーン
+	*/
+	void InitGameLib(char* title, int width, int height,HINSTANCE hInstance, LRESULT CALLBACK WndProc(HWND hWnd, UINT msg, WPARAM wp, LPARAM lp), bool windowType);
```
引数が長すぎて見辛いです。
WndProcは関数ポインタですよね？普通、hWnd等の仮引数は書かないです。
あと、インスタンスハンドルを取得するWin32APIがあった気がするので引数で渡さなくても良いかもです。 !fix

Sharoku-haga replied:
関数ポインタの仮引数はなくしました。
インスタンスハンドルですが、調べたらGetModuleHandle関数で取得できるとのことなので引数からなくし、関数内で取得する形にしました。

taishi-matsumura replied:
これでもまだ長くてイヤですが仕方無いですね。 !fixed

taishi-matsumura commented on Twar/Lib_Class/GameLib/Lib/GraphicsDevice.h:
```diff
@@ -0,0 +1,103 @@
+class GraphicsDevice
+{
+private:
+	HWND					m_hWnd;			//!< ウィンドウハンドル
```
メンバで保持する必要ありますか？ !fix

Sharoku-haga replied:
必要ないです。
フルスクリーンにするために試行錯誤していたときにメンバに加えて消し忘れてました。
すみません。修正しておきます。

Commit:
GameLibのInitGame関数修正

-インスタンスハンドルをなくし、GetModuleHandleによる関数内取得に変えました
-引数の関数の仮引数をなくしました

Commit:
必要ないメンバ変数削除と一部ソース修正

-ウィンドウ情報を保持するメンバ変数を削除
-フルスクリーンのパラメーターを修正

Yottomo commented on Twar/Lib_Class/GameLib/GameLib.h:
```diff
@@ -0,0 +1,348 @@
+	* @param[in] height ウインドウの縦幅
+	* @return GameLibクラス
+	*/
+	static GameLib& GetInstance()
```
GameLib.hのGetInstanceの
コメント間違ってない？

Sharoku-haga replied:
すみません。引数消すの忘れてました。後、Singletonの綴りも間違ってます。
修正しておきます。

Commit:
コメント修正

Yottomo commented on Twar/Lib_Class/GameLib/GameLib.h:
```diff
@@ -0,0 +1,345 @@
+﻿/**
```
DrawFontやStartTimer等の
デバッグ用の関数はデバッグ用であることが分かるように関数名を変更するか、コメントを記述して下さい。

Sharoku-haga replied:
デバック用のフォントの関数は普通のフォントクラスとの違いを明確に分ける為、関数名をDrawDebugFontに直しました。
デバック用の時間計測関連はDebugをつけると関数名が長くなるのでコメントにてデバック用であることを明記する形で修正しております。

Yottomo commented on Twar/Lib_Class/GameLib/GameLib.h:
```diff
@@ -0,0 +1,348 @@
+	* @param[in] height ウインドウの縦幅
+	* @return GameLibクラス
+	*/
+	static GameLib& GetInstance()
```
!fix

Yottomo replied:
status設定間違えました。
修正確認しました。
お疲れ様です。 !fixed

Commit:
デバック関連の修正

-デバック用フォント関数の名前を修正
-デバック用の時間計測関数関連にコメント追加

Yottomo commented on Twar/Lib_Class/GameLib/GameLib.h:
```diff
@@ -0,0 +1,345 @@
+﻿/**
```
修正確認しました。
お疲れ様です。 !fixed

taishi-matsumura commented on Twar/Lib_Class/GameLib/Lib/GraphicsDevice.h:
```diff
@@ -0,0 +1,103 @@
+class GraphicsDevice
+{
+private:
+	HWND					m_hWnd;			//!< ウィンドウハンドル
```
!fixed

Moto0116 commented on Twar/Lib_Class/GameLib/Lib/DebugTimer.h:
```diff
@@ -0,0 +1,90 @@
+#include <iostream>
+#include <vector>
+#include "DebugFont.h"
+using namespace std;
```
usingディレクティブでstdを指定するのは良くないのでやめといたほうがいいです。
それと.hでusingディレクティブを使用すると、それをインクルードしたcppとhも影響を受けてしまうので気をつけてください。
[参考](http://ppp-lab.sakura.ne.jp/cpp/language/018.html)

どうしても使用したい場合はhではなくcppで、using宣言を使用してください。
[参考](http://www.geocities.jp/bleis_tift/cpp/namespace.html)

Sharoku-haga replied:
参考サイトの方確認致しました。
修正作業に入ります。

taishi-matsumura reviewed (changes requested):
@Moto0116 の指摘の件、Changes Request(ApproveされるまでMergeできない)にしておきます

Moto0116 commented on Twar/Lib_Class/GameLib/Lib/InputDevice.cpp:
```diff
@@ -36,87 +36,87 @@ InputDevice::~InputDevice()
 	{
-		MessageBox(0, "�����[�h�̐ݒ�Ɏ��s", NULL, MB_OK);
+		MessageBox(0, "軸モードの設定に失敗", NULL, MB_OK);
 		return false;
```
返り値間違っていませんか？

Sharoku-haga replied:
間違っています。
修正しておきました。

Commit:
namespaceにおける修正

-using宣言をしている場所を削除
-InputDeviceの一部戻り値の修正

Sharoku-haga commented:
@Moto0116 さんの指摘の件の修正完了しました。
いい機会ですので、namespaceに関して深く掘り下げて勉強してみます。
アドバイスありがとうございました。

Moto0116 commented on Twar/Lib_Class/Font/Font.h:
```diff
@@ -0,0 +1,70 @@
+	* @param[in] format	テキストタイプ DT_LEFT(左寄せ) or DT_RIGHT(右寄せ) or DT_BOTTOM(下寄せ) or DT_TOP(上寄せ)
+	* @param[in] color		文字の色
+	*/
+	void Draw(LPCSTR pString, D3DXVECTOR2 pos, DWORD format, D3DCOLOR color);
```
第一引数の型間違えていませんか？
一つ上で定義しているDraw関数はLPCTSTRでしたけど。

Sharoku-haga replied:
間違ってます。
修正しておきました。

Commit:
引数の型を修正

-フォントクラスとデバックフォントクラスの一部引数を修正

Commit:
@Moto0116のアドバイスによる修正

一部ソース変更

Commit:
XFileの一部ソースを修正
